device/arista/x86_64-arista_7280cr3mk_32p4/Arista-7280CR3-C40/context_config.json

```diff
@@ -0,0 +1,157 @@
+{
+    "CONTEXTS": [
+        {
+            "guid" : 0,
+            "name" : "asic0",
+            "dbAsic" : "ASIC_DB",
+            "dbCounters" : "COUNTERS_DB",
+            "dbFlex": "FLEX_COUNTER_DB",
+            "dbState" : "STATE_DB",
+            "zmq_enable" : false,
+            "zmq_endpoint": "tcp://127.0.0.1:5565",
+            "zmq_ntf_endpoint": "tcp://127.0.0.1:5566",
+            "switches": [
+                {
+                    "index" : 0,
+                    "hwinfo" : ""
+                }
+            ]
+        },
+        {
+            "guid" : 1,
+            "name" : "phy1",
+            "dbAsic" : "GB_ASIC_DB",
```

Comment:
duplicated databases, please read here: https://github.com/Azure/sonic-sairedis/wiki/Context-configuration

Reply:
i also think what you intended is to have 2 syncd , first with ASIC_DB and second with GB_ASIC_DB, first will have 1 switch, and 2nd will have 8 phys

Reply:
@kcudnik : in this context_config.json file, trying to see/understand what's the purpose of having both ASIC_DB and GB_ASIC_DB blocks. Is it like this:
If a ASIC complex on the board has both ASIC and PHY on the same chip/SOC, then specify both of them.
If a PHY chip/device is Ext to ASIC (i.e. independent chip on the board), then just specifying GB_ASIC_DB block is sufficient?

Reply:
from my point of view, how hardware is organized is not relevant, depends how you want to present it in database and configuration file. you **CANT** have multiple syncd pointing to the same databases ex. ASIC_DB, each context entry with separate GUID is a separate syncd instance. please read context config wiki page

Reply:
@shyam77git , for this platform, guid 0 is for ASIC, guid 1 and thereafter are for external phys now. Later, we may have a further change to cover all external phys by guid 1

Reply:
@byu343, per this platform (this PR):
>> guid 1 and thereafter are for external phys now. 
Can you clarify now vs later? 
From what I gather from context config wiki page & from Kamil's comment here, multiple guid pointing to same database (e.g GB_ASIC_DB ) is not supported so how come this model works (i.e. different guid for each PHY)?

Reply:
@kcudnik I tried to set all phys to share the same context guid 1, and it has two new issues:

In https://github.com/Azure/sonic-swss/blob/master/orchagent/saihelper.cpp#L421, SAI_REDIS_SWITCH_ATTR_CONTEXT is set to phy_id, which is different for each phy. How about add a new property called context to gearbox_config.json, set it to 1 for all phys and then change line 421 to attr.value.u64 = phy->context.

In https://github.com/opencomputeproject/SAI/blob/master/inc/saiport.h#L748, SAI_PORT_ATTR_HW_LANE_LIST is defined as key. For ports with the same in-chip id but from different phy chips, they have the same SAI_PORT_ATTR_HW_LANE_LIST. After putting those ports in the same context, we encountered an error related to key conflict, which is from https://github.com/Azure/sonic-sairedis/blob/master/meta/Meta.cpp#L5060. I tried to keep SAI_PORT_ATTR_HW_LANE_LIST from being key in saiport.h#L748,  and it seems good. As that involves the change to SAI headers, is there any other better solution?

Reply:
this model works if you each syncd will have its own database

Reply:
we will not add any special properties like you proposed, context_config.json describes each syncd instance and that's it. its common for all vendors and architectures

Reply:
@shyam77git, between now vs later, maybe you can just skip the "now" model, as it does not follow the design in the wiki page and will very likely be deprecated. If you are still interested in why multiple guid pointing to same database could work, the tricky part is we did only create the syncd process for guid 1. It was done in that way to bypass some other issues.

Reply:
@kcudnik ,
The new property proposed is for gearbox_config.json, not for context_config.json.  Let me just create the new PR for it, to let it be clear in discussion.

Reply:
@kcudnik The code has been updated. 
For context_config.json, all the phys are defined in the same context. 
For gearbox_config.json, a property of context_id is added to each phy. It is to be used by gearsyncd in sonic-swss, as gearsyncd needs to specify the context for each phy when create_switch(): https://github.com/Azure/sonic-swss/pull/1826/files#diff-be825ae02c0d05e1a802a265c12721e5747035acab211f3bc7e6a219cf500e03L421.  phy_id in gearbox_config.json was used to specify the context which seems not always true. To support context_id in gearsyncd, a change to swss is created: https://github.com/Azure/sonic-swss/pull/1826.

Reply:
>> phy_id in gearbox_config.json was used to specify the context which seems not always true.

@byu343 
To elaborate more on this, share the understanding and ensure we all are on same page:
[1] Based on existing codebase in swss/saihelper, looks like each phy_id (in gearbox_config.json) corresponds to unique/separate guid (of context_config.json). whereas, with the change in context_config.json all PHYs (Switches) are under same guid (global context).
[2] At swss/sai code/workflow level, this change was done because phy_id was/is being passed as SAI_REDIS_SWITCH_ATTR_CONTEXT. 
[3] phy_id is unique for each PHY. However, context_config.json puts all PHYs (switches) under one (common) guid. 
[4] globalContext gets populated with phy_id for every create_switch() call which make globalContext different from every create_switch() call , whereas the expectation is to have a common (same) id for all PHYs/Switches under the guid 1 of the revised context_config.json. For this purpose, context_id field was introduced for all PHYs under the same roof (i.e. guid 1 GB_ASIC_DB of context_config.json) and populate with same value for all these PHYs.

@kcudnik - can you please share you thoughts/comments on this and related PR (1826)

<Excerpt from current codebase>
saihelper.cpp 
initSaiPhyApi() 
{
...
    /* Must be last Attribute */
    attr.id = SAI_REDIS_SWITCH_ATTR_CONTEXT;
    attr.value.u64 = phy->phy_id;
    attrs.push_back(attr);

    status = sai_switch_api->create_switch(&phyOid, (uint32_t)attrs.size(), attrs.data());
....
}

Sai.cpp 
sai_status_t Sai::create(
{
...
      if (attr_list[attr_count - 1].id == SAI_REDIS_SWITCH_ATTR_CONTEXT)
        {
            globalContext = attr_list[--attr_count].value.u32;
        }
...
}

Thanks,
Shyam

Reply:
> whereas the expectation is to have a common (same) id for all PHYs/Switches under the guid 1 of the revised context_config.json.

what do you mean to its expected to have common id ? which id are you talking about here?